scenarioo-server/src/main/java/org/scenarioo/business/diffViewer/ComparisonExecutor.java

```diff
@@ -60,99 +59,183 @@
+		LOGGER.info("Scheduling Comparison of build for background calculation: \n"
+			+ "     " + getComparisonConfigString(baseBranchName, baseBuildName, comparisonConfiguration));
 
-		return asyncComparisonExecutor.submit(new Callable<BuildDiffInfo>() {
+		Future<BuildDiffInfo> futureComparison = asyncComparisonExecutor.submit(new Callable<BuildDiffInfo>() {
 			@Override
 			public BuildDiffInfo call() {
-				return runComparison(baseBranchName, baseBuildName, comparisonConfiguration);
+				return calculateComparison(baseBranchName, baseBuildName, comparisonConfiguration);
 			}
 		});
-	}
 
-	private BuildDiffInfo runComparison(String baseBranchName, String baseBuildName,
-										ComparisonConfiguration comparisonConfiguration) {
-		ThreadLogAppender comparisonLog = null;
 		ComparisonParameters comparisonParameters = new ComparisonParameters(baseBranchName, baseBuildName, comparisonConfiguration,
-			configurationRepository.getConfiguration().getDiffImageAwtColor());;
-		BuildDiffInfo buildDiffInfo = null;
-		ComparisonConfiguration resolvedComparisonConfiguration = null;
+			configurationRepository.getConfiguration().getDiffImageAwtColor());
 
-		try {
-			comparisonLog = registerLogFile(baseBranchName, baseBuildName, comparisonConfiguration);
-			long startTime = System.currentTimeMillis();
+		storeComparisonScheduled(comparisonParameters);
 
-			LOGGER.info("=== START OF BUILD COMPARISON ===");
-			logBaseBuildAndComparisonConfiguration(baseBranchName, baseBuildName, comparisonConfiguration);
+		return futureComparison;
+	}
 
-			resolvedComparisonConfiguration = resolveComparisonConfiguration(comparisonConfiguration, baseBuildName);
+	private BuildDiffInfo calculateComparison(String baseBranchName, String baseBuildName,
+											  ComparisonConfiguration comparisonConfiguration) {
 
-			if (resolvedComparisonConfiguration == null) {
-				LOGGER.warn("No comparison build found for base build: " + baseBranchName + "/"
-					+ baseBuildName + " with defined comparison: " + comparisonConfiguration.getName());
-			} else {
-				comparisonParameters = new ComparisonParameters(baseBranchName, baseBuildName, resolvedComparisonConfiguration,
-					configurationRepository.getConfiguration().getDiffImageAwtColor());
-				storeComparisonInProgress(comparisonParameters);
-				buildDiffInfo = new UseCaseComparator(comparisonParameters).compare();
-				storeComparisonSuccessful(comparisonParameters, buildDiffInfo);
+			BuildDiffInfo buildDiffInfo;
+			ComparisonConfiguration resolvedComparisonConfiguration;
+			ThreadLogAppender comparisonLog = null;
+			ComparisonParameters comparisonParameters = new ComparisonParameters(baseBranchName, baseBuildName, comparisonConfiguration,
+				configurationRepository.getConfiguration().getDiffImageAwtColor());
+			long startTime = System.currentTimeMillis();
+
+			// Precondition check: do nothing if comparison was allready calculated since scheduling it for calculation.
+			try {
```

Comment:
Why is the try / catch block required here?

Reply:
because this is the calculation implementation that is not allowed to throw any exceptions. The impl must ensure all exceptions are logged in any case and silently return from that impl. Will document it accordingly.